dist/gcc-compatible/Hacl_P256.h

```diff
@@ -249,30 +312,48 @@ bool Hacl_P256_decompression_not_compressed_form(uint8_t *b, uint8_t *result);
 bool Hacl_P256_decompression_compressed_form(uint8_t *b, uint8_t *result);
 
 /*
- Input: a point buffer (internal representation: uint8[64]), 
+Convert raw to 65-byte uncompressed.
+
+This function effectively prepends a 0x04 byte.
+
+Input: a point buffer (internal representation: uint8[64]), 
  result: a point in not compressed form (uint8[65]).
 */
 void Hacl_P256_compression_not_compressed_form(uint8_t *b, uint8_t *result);
 
 /*
- Input: a point buffer (internal representation: uint8[64]), 
- result: a point in not compressed form (uint8[33]).
+Convert raw to 33-byte compressed.
+
+  Input: `b`, the pointer buffer in internal representation, of type `uint8[64]`
+  Output: `result`, a point in compressed form, of type `uint8[33]`
+
 */
 void Hacl_P256_compression_compressed_form(uint8_t *b, uint8_t *result);
 
+
+/******************/
+/* ECDH agreement */
+/******************/
+
 /*
- Input: result: uint8[64], 
- scalar: uint8[32].
-  
- Output: bool, where True stands for the correct key generation. 
-  
- False means that an error has occurred (possibly that the result respresents point at infinity). 
-  
+Convert a private key into a raw public key.
+
+  Input: `scalar`, the private key, of type `uint8[32]`.
+  Output: `result`, the public key, of type `uint8[64]`.
+  Returns:
+  - `true`, for success, meaning the public key is not a point at infinity
+  - `false`, otherwise.
+
+  `scalar` and `result` MUST NOT overlap.
 */
 bool Hacl_P256_ecp256dh_i(uint8_t *result, uint8_t *scalar);
 
 /*
- 
+ECDH key agreement.
+
+This function takes a 32-byte secret key, another party's 64-byte raw public
```

Comment:
Public keys must be validated before use. I don't think that's happening in here so the comment should tell users to first validate the inputs.